src/model/learn.py

```diff
@@ -30,67 +29,88 @@ def create_multi_env(num_envs, env_kwargs):
+    print(f"Saving model to {model_save_path}.zip...")
+    model.save(model_save_path, exclude=["policy_kwargs"])
+    print("Done saving model")
+    return model_save_path
+
+def load_latest_file(model, env):
+    fileIndex = find_highest_file_index(model.__class__.__name__)
+    if fileIndex == -1:
+        print("There is no existing model to load, starting learning from scratch")
+    else:
+        model_load_path = f"./models/{model.__class__.__name__}/{fileIndex}"
+        assert os.path.exists(f"{model_load_path}.zip")
+        print(f"Loading existing model from {model_load_path}.zip...")
+        model.load(model_load_path, env=env)
+        print("Done loading model")
 
 def learn():
-    best = 0
-    j = 0
+    best_result = 0
+    best_file_path = None
 
-    episode_duration_seconds = 60
-    simulation_step_duration_msec = 50
-    steps_per_episode = math.ceil(
-        (episode_duration_seconds * 1000) / simulation_step_duration_msec
-    )
-    episodes_per_training_epoch = 500
     verbose = bool(int(os.environ.get("VERBOSE", 0)))
+    environment_count = os.environ.get("ENVIRONMENT_COUNT", 16)
+    episode_duration_seconds = os.environ.get("EPISODE_DURATION_SECONDS", 60)
+    simulation_step_duration_msec = os.environ.get("SIMULATION_STEP_DURATION_MSEC", 50)
+    episodes_per_training_iteration = os.environ.get("EPISODES_PER_TRAINING_ITERATION", 400)
+    training_iteration_count = os.environ.get("TRAINING_ITERATION_COUNT", 60)
+    reward_type = os.environ.get("REWARD_TYPE", "delta_damage")
+    steps_per_episode = math.ceil((episode_duration_seconds * 1000) / simulation_step_duration_msec)
+    env_kwargs = dict(
+        sim_duration_seconds=episode_duration_seconds,
+        sim_step_duration_msec=simulation_step_duration_msec,
+        reward_type=reward_type,
+        verbose=verbose,
+    )
+    env = initialize_environment(environment_count, env_kwargs)
+    model = initialize_model(env, verbose)
 
-    for i in range(60):
-        env_kwargs = dict(
-            sim_duration_seconds=episode_duration_seconds,
-            sim_step_duration_msec=simulation_step_duration_msec,
-            reward_type="delta_damage",
-            verbose=verbose,
-        )
-        env = create_multi_env(16, env_kwargs)
-        # env = create_single_env(env_kwargs)
-        model = MaskedDQN(MaskedPolicy, env, verbose=verbose)
-
-        model_load_path = f"./models/{j}-{model.__class__.__name__}"
-        if os.path.exists(f"{model_load_path}.zip"):
-            print("Loading existing model...")
-            model.load(model_load_path, env=env)
-            print("Done loading model")
-
-        model.learn(
-            total_timesteps=(steps_per_episode * episodes_per_training_epoch),
-            progress_bar=True,
-        )
+    for i in range(training_iteration_count):
+        model.learn(total_timesteps=(steps_per_episode * episodes_per_training_iteration), progress_bar=True)
```

Comment:
I don't this is really doing what we would expect. With these changes it would continue learning using the same model so it's not training a new model. I think we should remove this entire loop and just focus on training one model since that is what we should be optimizing for and figuring out.

Reply:
Training the same model is what I intended. I'm removing it for now, though, to keep things simple. I do think we're going to want to add something like this back later. Training can take a while, and we'll want to checkpoint the process with saves and evaluation just to see how it's progressing. It should be noted that the code it still loading the latest file with a matching model name and proceeding from there.

Reply:
I just added model name handling to the code so we can train across multiple sessions while keeping different models separate by name.